language/vm/src/views.rs

```diff
@@ -177,6 +178,17 @@ impl<'a, T: ModuleAccess> ModuleView<'a, T> {
     pub fn id(&self) -> ModuleId {
         self.module.self_id()
     }
+
+    /// Return the `StructHandleIndex` that corresponds to the normalized type `t` in this module's
+    /// table of `StructHandle`'s. Returns `None` if there is no corresponding handle
```

Comment:
I think most of these changes wouldn't be needed if we have a better `GlobalSignatureToken`. The thing is that the spec compiler never works with a single VerifiedModule but always can make use of all VerifiedModules in the dependency relation.

Reply:
The (poorly named) `TypeTag` type may be suitable for this.